core/chains/evm/headtracker/head_tracker.go

```diff
@@ -39,25 +41,67 @@ var (
 // HeadsBufferSize - The buffer is used when heads sampling is disabled, to ensure the callback is run for every head
 const HeadsBufferSize = 10
 
-type headTracker struct {
+type headTracker[
```

Comment:
This struct, and the NewHeadTracker() are now completely generalized.
So you could move them out to common/headtracker package in head_tracker.go file.
This file can only keep the evm... variants, and renamed to evm_head_tracker.go

Feel free to take this up in another PR if you like.

Reply:
thanks for this note, will be moving them into another PR. Thanks for the note!